topostats/tracing/disordered_tracing.py

```diff
@@ -0,0 +1,708 @@
+            'method_outlier'.
+        n_grain : int
+            Grain number being processed (only  used in logging).
+        """
+        self.image = image * 1e-9 if convert_nm_to_m else image
+        self.mask = mask
+        self.filename = filename
+        self.pixel_to_nm_scaling = pixel_to_nm_scaling * 1e-9 if convert_nm_to_m else pixel_to_nm_scaling
+        self.min_skeleton_size = min_skeleton_size
+        self.mask_smoothing_params = (
+            mask_smoothing_params
+            if mask_smoothing_params is not None
+            else {"gaussian_sigma": None, "dilation_iterations": 2}
+        )
+        self.skeletonisation_params = (
+            skeletonisation_params if skeletonisation_params is not None else {"method": "zhang"}
+        )
+        self.pruning_params = pruning_params if pruning_params is not None else {"method": "topostats"}
+        self.n_grain = n_grain
+        # Images
+        self.smoothed_mask = np.zeros_like(image)
+        self.skeleton = np.zeros_like(image)
+        self.pruned_skeleton = np.zeros_like(image)
+        # Trace
+        self.disordered_trace = None
+
+        # suppresses scipy splining warnings
+        warnings.filterwarnings("ignore")
+
+        LOGGER.debug(f"[{self.filename}] Performing Disordered Tracing")
+
+    def trace_dna(self):
+        """Perform the DNA skeletonisation and cleaning pipeline."""
+        # LOGGER.info(f"[{self.filename}] : mask_smooth_params : {self.mask_smoothing_params=}")
+        self.smoothed_mask = self.smooth_mask(self.mask, **self.mask_smoothing_params)
+        self.skeleton = getSkeleton(
+            self.smoothed_mask,
+            self.mask,
+            method=self.skeletonisation_params["method"],
+            height_bias=self.skeletonisation_params["height_bias"],
+        ).get_skeleton()
+        self.pruned_skeleton = prune_skeleton(self.smoothed_mask, self.skeleton, **self.pruning_params.copy())
+        self.pruned_skeleton = self.remove_touching_edge(self.pruned_skeleton)
+        self.disordered_trace = np.argwhere(self.pruned_skeleton == 1)
+
+        if self.disordered_trace is None:
+            LOGGER.info(f"[{self.filename}] : Grain failed to Skeletonise")
+        elif len(self.disordered_trace) < self.min_skeleton_size:
+            self.disordered_trace = None
+
+    def re_add_holes(
+        self, orig_mask: npt.NDArray, new_mask: npt.NDArray, holearea_min_max: list = (4, np.inf)
+    ) -> npt.NDArray:
+        """
+        Restore holes in masks that were occluded by dilation.
+
+        As Gaussian dilation smoothing methods can close holes in the original mask, this function obtains those holes
+        (based on the general background being the first due to padding) and adds them back into the smoothed mask. When
+        paired with smooth_mask, this essentially just smooths the outer edge of the mask.
```

Comment:
Markup for inline code...

```suggestion
        paired with ``smooth_mask``, this essentially just smooths the outer edge of the mask.
```

Reply:
Didn't realise you could use markup in docstrings.

Reply:
Yep, docstrings are written in [reStructured Text](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html).